File: 2406-divide-intervals-into-minimum-number-of-groups/2406-divide-intervals-into-minimum-number-of-groups.cpp
class Solution {
public:
    int minGroups(vector<vector<int>>& intervals) {
        priority_queue<int,vector<int>,greater<int>> d;
        sort(intervals.begin(),intervals.end());
        int i=0;
        int ans=0;
        while(i<intervals.size()){
            if(d.size()==0 || d.top()>=intervals[i][0]){
                ans+=1;
                d.push(intervals[i][1]);
            }
            else{
                d.pop();
                d.push(intervals[i][1]);
                
            }
            i+=1;
        }
        return ans;
    }
};

File: 2406-divide-intervals-into-minimum-number-of-groups/2406-divide-intervals-into-minimum-number-of-groups.py
from queue import PriorityQueue
class Solution:
    def minGroups(self, intervals: List[List[int]]) -> int:
        intervals.sort()
        # print(intervals)
        obj=PriorityQueue()
        ans=0
        for item in intervals:
            if(obj.qsize()==0):
                obj.put(item[1])
                ans+=1
            else:
                a=obj.get()
                if(item[0]<=a):
                    ans+=1
                    obj.put(a)
                    obj.put(item[1])
                else:
                    obj.put(item[1])
        return ans
                        